account_operating_unit/models/account_invoice.py

```diff
@@ -18,6 +18,26 @@ class AccountInvoice(models.Model):
                                         states={'draft': [('readonly',
                                                            False)]})
 
+    @api.onchange('operating_unit_id')
+    def _onchange_operating_unit(self):
+        if self.operating_unit_id:
+            res = self.env['account.journal'].search(
+                [('type', '=', self.journal_id.type)])
+            rf = res.filtered(
+                    lambda aj: aj.operating_unit_id == self.operating_unit_id)
+            if not rf:
+                self.journal_id = res[0]
+            else:
+                self.journal_id = rf[0]
+
+    @api.onchange('journal_id')
+    def _onchange_journal(self):
+        if self.journal_id:
```

Comment:
`if self.journal_id and self.journal_id.operating_unit_id and self.journal_id.operating_unit_id != self.operating_unit_id'